pkg/terraform/configmap.go

```diff
@@ -46,18 +35,20 @@ func CreateConfigMap(name string, namespace string, ws *terraformv1alpha1.Worksp
 }
 
 func formatBackendTerraform(ws *terraformv1alpha1.Workspace) string {
-	backendType := ws.Spec.Backend.Type
-	bucket := ws.Spec.Backend.Bucket
-	key := ws.Spec.Backend.Key
-	region := ws.Spec.Backend.Region
-	dbTable := ws.Spec.Backend.DynamoDBTable
-
-	backend := fmt.Sprintf(backendTemplate, backendType, bucket, key, region, dbTable)
+	var region string
+	if ws.Spec.Region == "cn-north-1" || ws.Spec.Region == "cn-northwest-1" {
+		region = "cn-north-1"
+	} else {
+		region = "us-west-2"
+	}
+	backend := fmt.Sprintf(BackendTemplate, key, region, ws.Namespace)
```

Comment:
The key needs to include the workspace name as well, for example: `s3://bucket_name/namespace_name/workspace_name/terraform.tfstate`.

Reply:
Terraform does this. With the `workspace_key_prefix` set, it will automatically put in our namespace. Then within the context of the workspace, it creates the workspace name as part of the path. If we want an additional path, we can, but at the moment if someone creates `bar-network` in `foo` namespace, it will create the appropriate path as `bucket_name/foo/bar-network/terraform.tfstate`.

Reply:
Doh, thanks for the reminder. Carry on.